src/engine/x86-64/X86_64Runtime.v3

```diff
@@ -234,6 +223,33 @@ component X86_64Runtime {
 		stack.rsp += -Pointer.SIZE;
 		return null;
 	}
+	// Used exclusively for {resume_throw} to throw without overwriting %rsp.
+	// This function is hacky in that it is called with {curStack.parent}'s
+	// context (to prevent transferring control flow to {curStack}), but uses the
+	// {rsp} of {curStack} to ensure stack unwinding works correctly.
+	def runtime_THROW_DIRECT(instance: Instance, tag_index: u31) -> Throwable {
```

Comment:
This sounds a little dangerous and I think we should find a more robust solution. In particular, I don't understand what's special about a the parent of the stack that is doing a `resume_throw`. It seems like this should take the target continuation.